client/app/visualizations/chart/ChartRenderer.jsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { compact, each, extend, findKey, includes, fromPairs, get, map, sortBy, values } from 'lodash';
+
+import { ClientConfig, QueryData } from '@/components/proptypes';
+import PlotlyChart from './PlotlyChart';
+
+function getSeriesNames(mapping, columns) {
+  return compact(map(sortBy(columns, 'zIndex'), col => includes(['y', 'series', 'multiFilter', 'multi-filter'], mapping[col.name]) && col.name));
+}
+
+function addPointToSeries(point, seriesCollection, seriesName) {
+  if (seriesCollection[seriesName] === undefined) {
+    seriesCollection[seriesName] = {
+      name: seriesName,
+      type: 'column',
+      data: [],
+    };
+  }
+
+  seriesCollection[seriesName].data.push(point);
+}
+
+function chartData(mapping, data) {
+  const series = {};
+
+  data.rows.forEach((row) => {
+    let point = { $raw: row };
+    let seriesName;
+    let xValue = 0;
+    const yValues = {};
+    let eValue = null;
+    let sizeValue = null;
+    let zValue = null;
+
+    each(row, (v, definition) => {
+      definition = '' + definition;
+      const definitionParts = definition.split('::') || definition.split('__');
+      const name = definitionParts[0];
+      const type = mapping ? mapping[definition] : definitionParts[1];
+      let value = v;
```

Comment:
Can you rename `v` to `value` in the iterator, or are arguments treated as `const`?

Reply:
Hi @brylie! This PR is going to be abandoned - we already implemented a part of it in #3493 and will migrate other visualizations one by one for better testability (@washort did a really good job here, but he decided to migrate all at once, and it was too hard to test everything and fix bugs. we keep this PR as an example for a future smaller PRs).